### DISCUSSION TITLE
rb scripts: get dependency version

### DISCUSSION THREAD

**Discussion by sataset** [Homebrew/discussions#3648] [Casks] [answered]
Hi! So I have created a [merge request](https://github.com/Homebrew/homebrew-cask-drivers/pull/3004) for cask update and wondered if it is possible to get dependency version in order to choose which version to install?

Basically I am wondering if something like this possible:
```ruby
cask "droidcam-obs" do
  if obs.version >= 28
    version "1.6.0"
    sha256 "040e80cfe5469ae98f50206f58a218bc9efed36b11cafdc447659ed42e37988e"
  else
    version "1.5.0"
    sha256 "ae1ccf28b998f3cc8a7cbe63d43973cd3bc90a7abb590c6ee23cca278190090b"
  end

...

depends_on cask: "obs"

...
```

**Comment by SMillerDev** [comment, marked as answer]
Technically it should be possible. It's not really in line with brews behavior as a package manager though. Brew will always try and update everything.

**Reply by sataset** [reply]
Thank you for reply!

I think that is the main feature of most package managers? Dependency is not only about package itself but also a version that it is compatible with. Same as with macOS version, dependency package is no different, at least for me.

I tried searching cookbooks and did not find if it was possible to use this kind of code. Are there any ways to test it locally?

**Reply by SMillerDev** [reply]
Well, I meant Homebrew only supports the latest versions of all dependencies so this situation should never occur.

I just looked but unfortunately it does not seem possible to do this in a cask. For formulae we can do `Formula['go'].version`, but casks have no such lookup mechanism.
